src/Disks/ObjectStorages/AzureBlobStorage/AzureObjectStorage.h

```diff
@@ -29,18 +29,21 @@ struct AzureObjectStorageSettings
         uint64_t max_single_part_upload_size_,
         uint64_t min_bytes_for_seek_,
         int max_single_read_retries_,
-        int max_single_download_retries_)
+        int max_single_download_retries_,
+        int list_object_keys_size_)
         : max_single_part_upload_size(max_single_part_upload_size_)
         , min_bytes_for_seek(min_bytes_for_seek_)
         , max_single_read_retries(max_single_read_retries_)
         , max_single_download_retries(max_single_download_retries_)
+        , list_object_keys_size(list_object_keys_size_)
     {
     }
 
     size_t max_single_part_upload_size; /// NOTE: on 32-bit machines it will be at most 4GB, but size_t is also used in BufferBase for offset
     uint64_t min_bytes_for_seek;
     size_t max_single_read_retries;
     size_t max_single_download_retries;
+    int list_object_keys_size;
```

Comment:
Why is it `int` and not `size_t`? Why have you added that field?

Reply:
For two reasons: it was `int` before, and both (azure and aws) sdk has `int` for this parameter - https://github.com/ClickHouse/ClickHouse/pull/42950#discussion_r1019348213